### PR TITLE
chore: Fix CI/CD command utilities

### PR DESCRIPTION
**Pull request by palpatim**
The CI/CD command runner would periodically swallow errors instead of
propagating them, falsely reporting a successful pod push when it had in fact
failed. This change refactors the run_command utility to accurately reflect
command exit statuses so the CI/CD system will properly report failures.
In addition, this change includes formatting & linting fixes suggested by
black and flake8.

Although the run_integrationtests.py file uses the same utilities, refactoring
that is too large a task to take on with this fix.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by palpatim** [comment]
To do prior to marking PR as "Ready for review":

- Test all affected flows, fix regressions
  - Known issue: bump_sdk_version: passing the regex as a command argument double-escapes backslashes, making the `sed` command fail

**Comment by palpatim** [comment]
Tested scripts locally

**Comment by palpatim** [comment]
Unable to test `cocoapods_release.py` script since it pushes to the repo. Tested invocation flow by replacing `trunk push` with `lib lint`, which at least goes through the build process for the pods. To ensure error handling is working correctly, I manually introduced a build error in one of the pods and verified that the return value from the script was non-zero.

File: CircleciScripts/bump_sdk_version.py [hunk 1]
```diff
@@ -141,32 +140,32 @@ def bump_services(self):
...
+            ]
+        )
+        replace_files(self._root, service_pattern)
 
     def bump_podspecs(self):
         podspec_pattern1 = {
-            "enclosemark": "double",
             "exclude": "AWSCognitoIdentityProviderASF",
-            "match": r"(dependency[[:space:]]+'AWS.+'[[:space:]]*,[[:space:]]*')[0-9]+\.[0-9]+\.[0-9]+(')",
+            "match": r"(dependency[[:space:]]+'AWS.+'[[:space:]]*,[[:space:]]*')[0-9]+\.[0-9]+\.[0-9]+(')",  # noqa: E501
```
[the remark this answers is not in the document]
Comment (lawmicha) [reply CircleciScripts/bump_sdk_version.py:163]: oh line too long, https://stackoverflow.com/questions/47876079/how-to-tell-flake8-to-ignore-comments

File: CircleciScripts/bump_iossample_version.py [hunk 2]
```diff
@@ -1,48 +1,43 @@
...
+        "match": r"(pod[[:space:]]+'AWS.+'[[:space:]]*,[[:space:]]*'~>[[:space:]]*)[0-9]+\.[0-9]+\.[0-9]+'",  # noqa: E501
+        "replace": r"\1[version]'",
+        "files": [
             "CognitoAuth-Sample/Objective-C/Podfile",
             "CognitoAuth-Sample/Swift/Podfile",
             "CognitoYourUserPools-Sample/Objective-C/Podfile",
             "CognitoYourUserPools-Sample/Swift/Podfile",
             "IoT-Sample/Swift/Podfile",
             "Lex-Sample/ObjC/Podfile",
-            "Lex-Sample/swift-3/Podfile",
```
[the remark this answers is not in the document]
Comment (palpatim) [reply CircleciScripts/bump_iossample_version.py:21]: It doesn't exist. :)

**Pull request merged**